Formula/m/mumps.rb

```diff
@@ -0,0 +1,72 @@
+class Mumps < Formula
+  desc "MUltifrontal Massively Parallel sparse direct Solver"
+  homepage "https://mumps-solver.org/"
+  url "https://mumps-solver.org/MUMPS_5.6.2.tar.gz"
+  sha256 "13a2c1aff2bd1aa92fe84b7b35d88f43434019963ca09ef7e8c90821a8f1d59a"
+  license "CECILL-C"
+
+  # Core dependencies
+  depends_on "gcc"
+  depends_on "openblas"
+
+  def install
```

Comment:
There is so much magic here, can you try and upstream as much as possible?

Reply:
Are you referring to the custom installation logic? The upstream build system does not provide such a target. I can attempt to contribute one upstream (although MUMPS does not make this process obvious) but I'd rather not hold up this PR for MUMPs annual releases....

Reply:
We can use patches if they're submitted upstream, but without that it'll be much harder to maintain this in the long run

Reply:
To be clear, the only "magic" we're discussing is the installation phase?

Reply:
Yeah, setting a bunch of variables that could be set from upstream like: `optf = ["OPTF=-O"]` for example

Reply:
I've removed a number of these changes, including the one refered to above, but I'll respond w.r.t. to the additional comment make to this effect.

MUMPS has no public VCS that homebrew can link to to pull a patch. We're stuck waiting for a release to come out (and these changes would have to stay as is anyway), and then someone would have evaluate whether to keep these or not when the package is updated. 

I'll email a patch file to the upstream MUMPS mailing list, but I do not have the bandwidth to make a more pervasive update to their build system if they require a more thorough fix to the issues on top of the minimally functional patch.